### DISCUSSION TITLE
O2M/M2O fiels vs alias

### DISCUSSION THREAD

**Discussion by eliteproxy7** [directus/directus#10304] [Archive]
When I add a O2M or M2O the "Many" collection has a field in the table while the "One" collection does not .. and its added as an alias field in the directus_field

is there a way to have both M & O collections have the related fields? this way I can set a webhook actions on the O collection if I when the M collections are updated

**Comment by joselcvarela** [comment]
Hello @eliteproxy7 
For your use case I recommend to create another M2O field so you would use it for your O2M relation.
In the end, you would have 

Collection A:
1. M2O field for your main purpose
2. M2O field to link to O2M to Collection B

Collection B:
1. O2M field to link to A.2

M2O / O2M are direct relationships, i.e. they don't need a middle table for linking. Because of that, O2M fields are just an alias which only reads the other collection items M2O field to check if the current item ID is in there or not.
